File: app/src/main/java/org/jay/launchstarterexample/net/FrescoTraceListener.java
package org.jay.launchstarterexample.net;

import com.facebook.imagepipeline.listener.RequestListener;
import com.facebook.imagepipeline.request.ImageRequest;

import javax.annotation.Nullable;
import java.util.Map;

public class FrescoTraceListener implements RequestListener {

    @Override
    public void onRequestStart(ImageRequest request, Object callerContext, String requestId, boolean isPrefetch) {

    }

    @Override
    public void onRequestSuccess(ImageRequest request, String requestId, boolean isPrefetch) {

    }

    @Override
    public void onRequestFailure(ImageRequest request, String requestId, Throwable throwable, boolean isPrefetch) {

    }

    @Override
    public void onRequestCancellation(String requestId) {

    }

    @Override
    public void onProducerStart(String requestId, String producerName) {

    }

    @Override
    public void onProducerEvent(String requestId, String producerName, String eventName) {

    }

    @Override
    public void onProducerFinishWithSuccess(String requestId, String producerName, @Nullable Map<String, String> extraMap) {

    }

    @Override
    public void onProducerFinishWithFailure(String requestId, String producerName, Throwable t, @Nullable Map<String, String> extraMap) {

    }

    @Override
    public void onProducerFinishWithCancellation(String requestId, String producerName, @Nullable Map<String, String> extraMap) {

    }

    @Override
    public void onUltimateProducerReached(String requestId, String producerName, boolean successful) {

    }

    @Override
    public boolean requiresExtraMap(String requestId) {
        return false;
    }
}

File: app/src/main/java/org/jay/appstarterexample/net/FrescoTraceListener.kt
package org.jay.appstarterexample.net

import com.facebook.imagepipeline.listener.RequestListener
import com.facebook.imagepipeline.request.ImageRequest

class FrescoTraceListener : RequestListener {
    override fun onRequestStart(
        request: ImageRequest,
        callerContext: Any,
        requestId: String,
        isPrefetch: Boolean
    ) {
    }

    override fun onRequestSuccess(request: ImageRequest, requestId: String, isPrefetch: Boolean) {}
    override fun onRequestFailure(
        request: ImageRequest,
        requestId: String,
        throwable: Throwable,
        isPrefetch: Boolean
    ) {
    }

    override fun onRequestCancellation(requestId: String) {}
    override fun onProducerStart(requestId: String, producerName: String) {}
    override fun onProducerEvent(requestId: String, producerName: String, eventName: String) {}
    override fun onProducerFinishWithSuccess(
        requestId: String,
        producerName: String,
        extraMap: Map<String, String>?
    ) {
    }

    override fun onProducerFinishWithFailure(
        requestId: String,
        producerName: String,
        t: Throwable,
        extraMap: Map<String, String>?
    ) {
    }

    override fun onProducerFinishWithCancellation(
        requestId: String,
        producerName: String,
        extraMap: Map<String, String>?
    ) {
    }

    override fun onUltimateProducerReached(
        requestId: String,
        producerName: String,
        successful: Boolean
    ) {
    }

    override fun requiresExtraMap(requestId: String): Boolean {
        return false
    }
}